turbine/src/sigverify_shreds.rs

```diff
@@ -56,11 +60,12 @@ pub fn spawn_shred_sigverify(
             if deduper.maybe_reset(&mut rng, DEDUPER_FALSE_POSITIVE_RATE, DEDUPER_RESET_CYCLE) {
                 stats.num_deduper_saturations += 1;
             }
+            // We can't store the keypair outside the loop
+            // because the identity might be hot swapped.
```

Comment:
I'm not overly familiar with how identity hot swap gets synchronized across all these different services. Is it okay that we could be holding a stale copy while in `run_shred_sigverify`?

Reply:
The identity hot-swap updates `ClusterInfo::keypair` and most services rely on `ClusterInfo::keypair` to get synchronized.
We might be holding a stale copy in `run_shred_sigverify` but only for one iteration of this `loop` and I think that is fine given that 50% of shreds are redundant anyways and identity hot-swaps are rare and infrequent.

That said the identity hot-swap has been a PITA to deal with.

Reply:
Ah, I see. So it doesn't need to be perfectly synchronized. Worst case we drop a few shreds due to invalid signature during the transition.

Reply:
right.